tools/virtiostat.py

```diff
@@ -194,6 +222,29 @@
 }
 """
 
+# filter for driver name
+if args.driver:
+    bpf_text = bpf_text.replace('DRIVERFILTER',
+        """char filter_driver[] = \"%s\";
+        char driver[16];
+        bpf_probe_read_kernel(driver, sizeof(driver), vq->vdev->dev.driver->name);
+        if (local_strcmp(filter_driver, driver))
+        return;""" % (args.driver))
+else:
+    bpf_text = bpf_text.replace('DRIVERFILTER', '')
+
+# filter for dev name
+if args.devname:
+    bpf_text = bpf_text.replace('DEVNAMEFILTER',
+        """char filter_devname[] = \"%s\";
+        char devname[16];
+        bpf_probe_read_kernel(devname, sizeof(devname), vq->vdev->dev.kobj.name);
```

Comment:
bpf_probe_read_kernel_str()?

Reply:
bpf_probe_read_kernel_str is better to copy a string from kernel, so also replace other codes together.